cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -198,6 +198,44 @@ runTransactionBuildCmd
         first TxCmdProposalError
           <$> readTxGovernanceActions eon proposalFiles
 
+    -- Extract return addresses from proposals and check that the return address in each proposal is registered
+
+    let returnAddrHashes =
+          Set.fromList
+            [ StakeCredentialByKey returnAddrHash
+            | (proposal, _) <- proposals
+            , let (_, returnAddrHash, _) = fromProposalProcedure eon proposal -- fromProposalProcedure needs to be adjusted so that it works with script hashes.
+            ]
+        treasuryWithdrawalAddresses =
+          Set.fromList
+            [ stakeCred
+            | (proposal, _) <- proposals
+            , let (_, _, govAction) = fromProposalProcedure eon proposal
+            , TreasuryWithdrawal withdrawalsList _ <- [govAction] -- Match on TreasuryWithdrawal action
```

Comment:
As I said during the pair programming session, I don't like pattern matching in list comprehensions generally speaking. But in this case, it would be way more verbose to do differently, so I'm fine with this version.

Reply:
I was going to say I thought it was an interesting way of pattern-matching